.github/workflows/build-test-ship.yaml

```diff
@@ -183,34 +197,248 @@ jobs:
+              fi
+            done
+          done
+          echo Dependencies: ${!deps[@]}
+          echo "deps=${!deps[@]}" >> $GITHUB_OUTPUT
+
+      - name: Start postgres and create service databases and roles
+        shell: bash
+        run: |
+          docker-compose up -d postgres
+          timeout 1m bash -c 'until docker-compose run postgres pg_isready -d postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres; do sleep 1s; done'
+          declare -A dbs
+          conn_regex='postgres://(\w*):(\w*)@\$\{POSTGRES_SERVER\}/(\w*)'
+          for dependency in ${{ steps.parse-deps.outputs.deps }}; do
+            # Parse DB from POSTGRES_CONNECTION. Parse roles and passwords from sql/roles.sql
+            conn=$(yq ".services.${dependency}.environment.POSTGRES_CONNECTION" docker-compose.yml)
+            if [[ ${conn} =~ ${conn_regex} ]]; then
+              db=${BASH_REMATCH[3]}
+              if [[ ! ${dbs[${db}]} ]]; then
+                dbs[${db}]=1
+                docker exec -i -e PGPASSWORD=${POSTGRES_PASSWORD} postgres \
+                  psql -h 0.0.0.0 -U ${POSTGRES_USER} -a -c "CREATE DATABASE ${db}"
+                context=$(yq ".services.${dependency}.build.context" docker-compose.yml)
+                roles="${context}/sql/roles.sql"
+                if [[ -f ${roles} ]]; then
+                  docker compose cp ${roles} postgres:roles.sql
+                  docker exec -i -e PGPASSWORD=${POSTGRES_PASSWORD} postgres \
+                    psql -h 0.0.0.0 -U ${POSTGRES_USER} -d ${db} -a -f roles.sql
+                fi
+                extensions="${context}/sql/extensions.sql"
+                if [[ -f ${extensions} ]]; then
+                  docker compose cp ${extensions} postgres:extensions.sql
+                  docker exec -i -e PGPASSWORD=${POSTGRES_PASSWORD} postgres \
+                    psql -h 0.0.0.0 -U ${POSTGRES_USER} -d ${db} -a -f extensions.sql
+                fi
+              else
+                echo ${db} database already initialized
+              fi
+            fi
+          done
+
+      - name: Launch services with dependencies via docker-compose
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        shell: bash
+        run: |
+          docker-compose pull --include-deps ${{ steps.get-services.outputs.services }}
+          docker-compose up -d ${{ steps.get-services.outputs.services }}
+          sleep 1m
+
+      - name: Run API Proctor tests
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        shell: bash
+        run: |
+          docker run --rm --name api-proctor --network host \
+            -e READ_POSTGRES_CONNECTION=postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@localhost:${POSTGRES_PORT} \
+            -e WRITE_POSTGRES_CONNECTION=postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@localhost:${POSTGRES_PORT} \
+            -e NODE_ENV=ci \
+            $ECR_REGISTRY/api-proctor:latest \
```

Comment:
Excellent points! If I'm understanding your proposed solution, I think that would rely on naming the `api-proctor` branch the same as the service branch? That sounds workable, and I can't immediately think of a better way to associate the two. We could use a manual override, but then we'd have to remember to swap it back after the `api-proctor` branch is merged, which sounds error-prone.

Reply:
Yeah, that's about the shape of my proposed hack :sweat_smile: This is probably something that BE and SEIT would need to hash out jointly, I'd imagine. Is there still a regular CI meeting going on that this could be discussed in? Regardless, I think it would be just fine to move forward with this PR and figure out this "api-proctor branch override" issue at a later date, since it seems like we are pretty far away from enforcing CI passing as a blocker to merging to `dev`.

> We could use a manual override, but then we'd have to remember to swap it back after the api-proctor branch is merged, which sounds error-prone.

Totally agreed -- anything that requires checking something into the branch that needs to be reverted on merge to `dev` seems worthwhile to avoid to me.

Reply:
This has maybe been discussed elsewhere but is there a compelling reason that API proctor test cases can't live in the service repos they test? I would assume this is the usual pattern for CI to avoid needing to sync branches across repos for test-breaking changes.

Reply:
I'm very interested in Dizzy's suggestion. 👀 For now I've added a [backlog task](https://app.asana.com/0/1201010009456007/1203467942453227/f) to track the problem and possible solutions. Thanks again for bringing this up. 👍

Reply:
:d'oh: Great point @dzbee ! It worked for playwright, so surely it can work for `api-proctor` as well :P